src/main/scala/scalaz/distributed/Metadata.scala

```diff
@@ -0,0 +1,14 @@
+package scalaz.distributed
+
+trait Metadata[Type[_], Path[_, _]] {
+  def get[A: Type, B: Type](where: Path[A, B]): F[B]
+  def set[A: Type, B: Type](where: Path[A, B], b: B): F[Unit]
+
+  final def modify[A: Type, B: Type](where: Path[A, B], f: B => B): F[Unit] =
```

Comment:
This won't be atomic... instead of `modify` we may have to introduce type-specific operations based on the implementation. e.g. for an `Int` we can do atomic increment / decrement using CRDTs.